fl4health/model_bases/pca.py

```diff
@@ -0,0 +1,166 @@
+from logging import INFO
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from torch import Tensor
+from torch.nn.parameter import Parameter
+
+
+class PCAModule(nn.Module):
+    def __init__(self, low_rank: bool = True, full_svd: bool = False, rank_estimation: int = 6) -> None:
+        """
+        PyTorch module for performing Principal Component Analysis.
+
+        Args:
+            low_rank (bool, optional): Indicates whether the data matrix is low-rank. If the user has
+             prior knowledge that it is, then this parameter can be set to True to allow for more efficient
+              computation of SVD. Defaults to True.
+            full_svd (bool, optional): Indicates whether full SVD or reduced SVD is performed. Defaults to False.
+            rank_estimation (int, optional): An estimation of the rank of the data matrix.
+            Only used if self.low_rank is True. Defaults to 6.
+        """
+        super().__init__()
+        self.low_rank = low_rank
+        self.full_svd = full_svd
+        self.rank_estimation = rank_estimation
+        self.principal_components: Parameter
+        self.singular_values: Parameter
+        self.data_mean: Tensor
+
+    def forward(self, X: Tensor, center_data: bool) -> Tuple[Tensor, Tensor]:
+        """
+        Perform PCA on the data matrix X by computing its SVD.
+
+        Args:
+            X (Tensor): Data matrix.
+            center_data (bool): If true, then the data mean will be subtracted from all data points prior to
+            performing PCA.
```

Comment:
I would also state that if `center_data` is false, it is expected that the data is already centered and an exception will be thrown if it is not.